substrate/client/transaction-pool/src/fork_aware_txpool/view_store.rs

```diff
@@ -907,4 +909,39 @@ where
 			}
 		}
 	}
+
+	/// Searches in the store transactions provides tags.
+	///
+	/// Will use the set of block hashes to get associated views where the transactions provides
+	/// tags are searched.
+	pub(crate) fn provides_tags_from_views(
+		&self,
+		mut xts_hashes: Vec<ExtrinsicHash<ChainApi>>,
+		blocks_hashes: Vec<Block::Hash>,
+	) -> HashMap<ExtrinsicHash<ChainApi>, Vec<Tag>> {
+		let mut provides_tags_map = HashMap::new();
+
+		blocks_hashes.into_iter().for_each(|bh| {
+			if let Some((view, _)) = self.get_view_at(bh, true) {
```

Comment:
if could be avoided:
```suggestion
		blocks_hashes.into_iter().filter_map(|bh| self.get_view_at(bh, true).map(|(view, _)| view))
for_each(|view| {
```